python/llm/src/bigdl/llm/transformers/models/llama.py

```diff
@@ -266,6 +272,32 @@ def check_flash_attention_available(query):
     return True
 
 
+def use_esimd_sdp(q_len, head_dim, query_states):
+    if head_dim != 128:
+        # esimd_sdp only support head_dim = 128 now
+        return False
+    elif q_len != 1:
+        # esimd_sdp only support rest token now
+        return False
+    elif query_states.device.type != "xpu":
+        # esimd_sdp only support GPU now
+        return False
+    elif query_states.dtype != torch.float16:
+        # esimd_sdp only has optimization for FP16 now
+        return False
+    else:
+        device_name = torch.xpu.get_device_name(query_states.device.index)
+        if device_name.startswith("Intel(R) Arc(TM)") or \
```

Comment:
others LGTM, but it seems MTL doesn't support lsc function, which is widely used in esimd kernel. The esimd kernel's performance will be poor without lsc function, so we can simply think MTL doesn't support esimd kernel. The igpu device name of MTL is `Intel(R) Arc(TM) Graphics`, we should distinguish it from Arc

Reply:
Use "Intel(R) Arc(TM) A" as current workaournd to distinguish MTL from Arc.
May change this part as we know how to locate MTL.